includes/rest/class-server.php

```diff
@@ -145,6 +146,55 @@ public static function authorize_activitypub_requests( $response, $handler, $req
 		return $response;
 	}
 
+	/**
+	 * Callback function to authorize each api requests
+	 *
+	 * @see WP_REST_Request
+	 *
+	 * @see https://www.w3.org/wiki/SocialCG/ActivityPub/Primer/Authentication_Authorization#Authorized_fetch
+	 * @see https://swicg.github.io/activitypub-http-signature/#authorized-fetch
+	 *
+	 * @param \WP_REST_Request $request  Request used to generate the response.
+	 *
+	 * @return bool Whether the request is authorized.
+	 */
+	public static function authorize_activitypub_permissions( $request ) {
+		if ( 'HEAD' === $request->get_method() ) {
+			return true;
+		}
+
+		/**
+		 * Filter to defer signature verification.
+		 *
+		 * Skip signature verification for debugging purposes or to reduce load for
+		 * certain Activity-Types, like "Delete".
+		 *
+		 * @param bool             $defer   Whether to defer signature verification.
+		 * @param \WP_REST_Request $request The request used to generate the response.
+		 *
+		 * @return bool Whether to defer signature verification.
+		 */
+		$defer = \apply_filters( 'activitypub_defer_signature_verification', false, $request );
+
+		if ( $defer ) {
+			return true;
+		}
+
+		if (
+			// POST-Requests are always signed.
+			'GET' !== $request->get_method() ||
```

Comment:
Would it make sense to specifically check for `POST` here, or to `\WP_REST_Server::EDITABLE`?

Reply:
I wanted to be on the save side, so I thought I would love to everything except `GET` :)

Reply:
We had an issue, because initially I only checked for `POST`, that's why I was a bit paranoid with this change :)